rre-dataset-generator/tests/unit/test_elasticsearch_search_engine.py

```diff
@@ -0,0 +1,104 @@
+    # search_engine.extract_documents_to_generate_queries, which contains requests.post, uses the monkeypatch
+    result = search_engine.fetch_for_query_generation(documents_filter=elasticsearch_config.documents_filter,
+                                                      doc_number=elasticsearch_config.doc_number,
+                                                      doc_fields=elasticsearch_config.doc_fields)
+    assert result[0] == Document(**mock_dict)
+
+def test_elasticsearch_search_engine_fetch_for_evaluation(monkeypatch, elasticsearch_config, mock_doc, mock_dict):
+    url = "https://fakeurl"
+    search_engine = ElasticsearchSearchEngine(url)
+
+    payload = json.dumps({"match_all": {}})
+
+    # apply the monkeypatch for requests.post to mock_post
+    monkeypatch.setattr(requests, "post",
+                        lambda *args, **kwargs: MockResponseElasticsearchEngine([mock_doc], status_code=200)
+                        )
+    # search_engine.extract_documents_to_evaluate_system, which contains requests.post, uses the monkeypatch
+    result = search_engine.fetch_for_evaluation(keyword="and",
+                                                query_template=elasticsearch_config.query_template,
+                                                doc_fields=elasticsearch_config.doc_fields)
+    assert result[0] == Document(**mock_dict)
+
+def test_elasticsearch_search_engine_negative_post_fetch_for_query_generation_EXPECTED_HttpError(monkeypatch, elasticsearch_config):
+    for status_code in [400, 401, 402, 403, 500]:
+        monkeypatch.setattr(requests, "post", lambda *args, **kwargs: MockResponseElasticsearchEngine([],
+                                                                                                    status_code=status_code))
+
+        search_engine = ElasticsearchSearchEngine("https://fakeurl")
+
+        with pytest.raises(HTTPError):
+            search_engine.fetch_for_query_generation(
+                documents_filter=elasticsearch_config.documents_filter,
+                doc_number=elasticsearch_config.doc_number,
+                doc_fields=elasticsearch_config.doc_fields
+            )
+
+
+def test_elasticsearch_search_engine_negative_post_fetch_for_evaluation_EXPECTED_HttpError(monkeypatch, elasticsearch_config):
+    for status_code in [400, 401, 402, 403, 500]:
+        monkeypatch.setattr(requests, "post", lambda *args, **kwargs: MockResponseElasticsearchEngine([],
+                                                                                                    status_code=status_code))
+
+        search_engine = ElasticsearchSearchEngine("https://fakeurl")
+
+        with pytest.raises(HTTPError):
+            search_engine.fetch_for_evaluation(
+                keyword="and",
+                query_template=elasticsearch_config.query_template,
+                doc_fields=elasticsearch_config.doc_fields
+            )
+
+def test_elasticsearch_search_engine_bad_url_EXPECTED_ValidationError():
```

Comment:
Initially we were using the word "EXPECT" (uppercased) as a separator but the upper case words are not standard in python so we are moving to "__expect__" (with the double underscore before and after the word expect)